doc/crypto/api/ops/pake.rst

```diff
@@ -954,23 +954,30 @@ Multi-part PAKE operations
         Active PAKE operation.
     .. param:: const psa_key_attributes_t * attributes
         The attributes for the new key.
-        This function uses the attributes as follows:
 
-        *   The key type is required.
+        The following attributes are required for all keys:
+
+        *   The key type.
             All PAKE algorithms can output a key of type :code:`PSA_KEY_TYPE_DERIVE` or :code:`PSA_KEY_TYPE_HMAC`.
             PAKE algorithms that produce a pseudo-random shared secret, can also output block-cipher key types, for example :code:`PSA_KEY_TYPE_AES`.
             Refer to the documentation of individual PAKE algorithms for more information.
-        *   The key size in ``attributes`` must be zero.
+        *   The key size in ``attributes`` must be zero --- the default value.
```

Comment:
Preexisting: the correct value should be allowed.

The intent is that if you obtain a key's attributes with `psa_get_key_attributes`, what you obtain should always be valid to create a similar key.

Reply:
In this case, I think I recall that we didn't want to try and construct a helper macro that could report the actual key size that would be returned.

But perhaps the absence of a macro to compute a value from a provided cipher-suite, is not a good enough reason to refuse to accept a correct value. A caller who knows precisely what cipher suite they are using could specify the known key size here - and it would be counter-intuitive if that failed.

So I think I agree with relaxing this.

Reply:
I have now permitted the correct size